crates/test_utils/src/agent.rs

```diff
@@ -0,0 +1,164 @@
+//! Test Utility helpers around agents and agent info.
+
+use kitsune2_api::*;
+use std::sync::{Arc, Mutex};
+
+/// A default test space id.
+pub const TEST_SPACE: SpaceId =
+    SpaceId(id::Id(bytes::Bytes::from_static(b"test-space")));
+
+/// The test signature bytes.
+pub const TEST_SIG: &[u8] = b"test-signature";
+
+/// Test Verifier considers only signatures with the bytes
+/// `b"test-signature"` to be valid.
+#[derive(Debug)]
+pub struct TestVerifier;
+
+impl agent::Verifier for TestVerifier {
+    fn verify(
+        &self,
+        _agent_info: &agent::AgentInfo,
+        _message: &[u8],
+        signature: &[u8],
+    ) -> bool {
+        signature == TEST_SIG
+    }
+}
+
+struct TestLocalAgentInner {
+    cb: Option<Arc<dyn Fn() + 'static + Send + Sync>>,
+    cur: DhtArc,
+    tgt: DhtArc,
+}
+
+/// A test local agent is generated from incrementing id keys,
+/// and always signs with the signature `b"test-signature"`.
+pub struct TestLocalAgent {
```

Comment:
Ahh, clearly not. This is the test agent. So I'm unclear about the use of the agent type above then